### PR TITLE
Release 3.4.0

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: src/homepage/_posts/2019-09-05-james-3.4.0.markdown [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+The Apache James developers are pleased to announce James server 3.4.0 release.
+
+Here are some points we worked on:
+
+### Added
+- Add in-memory docker image
+- Support of AWS S3 as blobstore
+- WebAdmin routes for mapping management (AddAddressMapping, AddRegexMapping, ListMappings), previously only manageable by CLI
```
Comment (blackheaven) [inline src/homepage/_posts/2019-09-05-james-3.4.0.markdown]: maybe add back-quotes

File: src/homepage/_posts/2019-09-05-james-3.4.0.markdown [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---
+layout: post
```
Comment (blackheaven) [inline src/homepage/_posts/2019-09-05-james-3.4.0.markdown]: do you have screenshot?

File: src/site/xdoc/download.xml [hunk 3]
```diff
@@ -540,32 +528,32 @@ is found <a href='https://www.apache.org/licenses/exports/'>here</a>.
...
-      <li><a href="https://mvnrepository.com/artifact/org.apache.james.protocols/protocols-imap/3.3.0">IMAP</a></li>
-      <li><a href="https://mvnrepository.com/artifact/org.apache.james.protocols/protocols-pop3/3.3.0">POP3</a></li>
-      <li><a href="https://mvnrepository.com/artifact/org.apache.james.protocols/protocols-managesieve/3.3.0">MANAGESIEVE</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-api/3.4.0">API</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-netty/3.4.0">NETTY</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-smtp/3.4.0">SMTP</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-lmtp/3.4.0">LMTP</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-imap/3.4.0">IMAP</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-pop3/3.4.0">POP3</a></li>
+      <li><a href="https://repo1.maven.org/maven2/org/apache/james/protocols/protocols-managesieve/3.4.0">MANAGESIEVE</a></li>
```
Comment (rouazana) [inline src/site/xdoc/download.xml]: the files should be hosted by Apache, no?

File: src/homepage/index.html [hunk 4]
```diff
@@ -105,14 +105,14 @@ <h2>Get Started With James</h2>
             <ul class="james-ul no-padding">
               <li class="post-template"><span class="long-arrow-right">&#8594;</span><span><b>
 WHAT WILL YOU TRY:</b><br>
-              <span>Here you will try James server v 3.3.0 thanks to an image.This James image has a default configuration
+              <span>Here you will try James server v 3.4.0 thanks to an image.This James image has a default configuration
```
Comment (Arsnael) [inline src/homepage/index.html]: missing space between sentences `image. This James`

File: src/site/xdoc/server/index.xml [hunk 5]
```diff
@@ -40,22 +40,22 @@
...
           </span>
         </p>
          <p>You can also have a look to the last stable version</p>
 
-        <p>Apache James Server 3.3.0 represents the leading edge of development. This code stream has many more
+        <p>Apache James Server 3.4.0 represents the leading edge of development. This code stream has many more
           features than the 2.3 code, but is not be as well tested in production. Reasonable
           configuration compatibility has been retained with 2.3.2.</p>
           
         <p>Apache James Server 3.1.0 required Java 1.8. A migration guide for
```
Comment (chibenwa) [inline src/site/xdoc/server/index.xml]: `Apache James Server 3.1.0 required Java 1.8.` and following versions?

File: src/site/xdoc/download.xml (hunk 3 above)
Comment (mbaechler) [reply src/site/xdoc/download.xml]: the source code not the compiled artifacts

Comment (chibenwa) [reply src/site/xdoc/download.xml]: Not sure about that I remember moving back things to dist not that long ago

Comment (mbaechler) [reply src/site/xdoc/download.xml]: btw It was on mvnrepository and now on maven.org, none of which belongs to Apache.

I propose to keep it that way unless we are asked to distribute it only by apache infrastructure.

**Comment by mbaechler** [comment]
Screenshot_2019-09-20 Apache James

**Comment by Arsnael** [comment]
Tests are :green_apple: 
```
Caused by: java.lang.IllegalStateException: no known node for /home/jenkins/build/workspace/261
```
test this please

**Comment by chibenwa** [comment]
```
[3407c1aeff5307210da509f522447160a723081e] Step 4/22 : ADD destination/james-server-app-3.4.0-SNAPSHOT-app.zip /root/james-server-app-3.4.0-SNAPSHOT-app.zip
[3407c1aeff5307210da509f522447160a723081e] ADD failed: stat /var/lib/docker/tmp/docker-builder251295731/destination/james-server-app-3.4.0-SNAPSHOT-app.zip: no such file or directory
```

**Comment by chibenwa** [comment]
(Just rebased to have :green_apple: tests)

File: src/site/xdoc/download.xml [hunk 6]
```diff
@@ -420,13 +408,13 @@ is found <a href='https://www.apache.org/licenses/exports/'>here</a>.
...
   
-    <p>Apache Mailet 3.3.0 is the latest stable version. </p>
+    <p>Apache Mailet 3.4.0 is the latest stable version. </p>
 
 
     <p>Sources:
-      <a href="https://www.apache.org/dyn/closer.lua/james/mailets/3.3.0/apache-mailet-api-3.3.0-sources.zip">(Jar)</a>
-      [<a href="https://apache.org/dist/james/mailets/3.3.0/apache-mailet-api-3.3.0-sources.zip.sha1">SHA-1</a>]
-      [<a href="https://apache.org/dist/james/mailets/3.3.0/apache-mailet-api-3.3.0-sources.zip.asc">PGP</a>]
+      <a href="https://repo1.maven.org/maven2/org/apache/james/apache-mailet-api/3.3.0/apache-mailet-api-3.4.0-sources.jar">(Jar)</a>
```
Comment (remk) [inline src/site/xdoc/download.xml]: Still `3.3.0` ?

Comment (mbaechler) [reply src/site/xdoc/download.xml]: good catch

File: src/homepage/_posts/2019-09-05-james-3.4.0.markdown [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+### Added
+- Add in-memory docker image
+- Support of AWS S3 as blobstore
+- WebAdmin routes for mapping management (`AddAddressMapping`, `AddRegexMapping`, `ListMappings`), previously only manageable by CLI
+- Metrics for Deleted Message Vault
+- More metrics for BlobStore (new delete & deleteBucket configuration)
+
+### Changed
+- (experimental) New implementation of the deleted message vault on top of an object store, not relying anymore on a
```
Comment (blackheaven) [inline src/homepage/_posts/2019-09-05-james-3.4.0.markdown]: s/webAdmin/WebAdmin/

**Comment by remk** [comment]
```
[61962e3546564b6a83aa7ecf6d52676f4418c495] Wrapped by: org.apache.james.server.task.json.JsonTaskSerializer$UnknownTaskException: org.apache.james.json.JsonGenericSerializer$UnknownTypeException: unknown type testTask
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at org.apache.james.server.task.json.JsonTaskSerializer.deserialize(JsonTaskSerializer.java:73)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at org.apache.james.task.eventsourcing.distributed.RabbitMQWorkQueue.executeTask(RabbitMQWorkQueue.java:120)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:350)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.onNext(FluxSubscribeOn.java:151)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at reactor.core.publisher.FluxCreate$BufferAsyncSink.drain(FluxCreate.java:793)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at reactor.core.publisher.FluxCreate$BufferAsyncSink.next(FluxCreate.java:718)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at reactor.core.publisher.FluxCreate$SerializedSink.next(FluxCreate.java:153)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at org.apache.james.task.eventsourcing.distributed.RabbitMQExclusiveConsumer.lambda$consumeExclusiveManualAck$2(RabbitMQExclusiveConsumer.java:109)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at com.rabbitmq.client.impl.recovery.AutorecoveringChannel$2.handleDelivery(AutorecoveringChannel.java:577)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at com.rabbitmq.client.impl.ConsumerDispatcher$5.run(ConsumerDispatcher.java:149)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at com.rabbitmq.client.impl.ConsumerWorkService$WorkPoolRunnable.run(ConsumerWorkService.java:104)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[61962e3546564b6a83aa7ecf6d52676f4418c495] 	at java.base/java.lang.Thread.run(Thread.java:834)
[61962e3546564b6a83aa7ecf6d52676f4418c495] [INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.129 s - in org.apache.james.task.eventsourcing.distributed.RabbitMQWorkQueueTest
[61962e3546564b6a83aa7ecf6d52676f4418c495] [INFO] 
[61962e3546564b6a83aa7ecf6d52676f4418c495] [INFO] Results:
[61962e3546564b6a83aa7ecf6d52676f4418c495] [INFO] 
[61962e3546564b6a83aa7ecf6d52676f4418c495] [ERROR] Failures: 
[61962e3546564b6a83aa7ecf6d52676f4418c495] [ERROR]   DistributedTaskManagerTest.listShouldBeEmptyWhenNoTasks 
[61962e3546564b6a83aa7ecf6d52676f4418c495] Expecting empty but was:<[TaskExecutionDetails{taskId=TaskId{value=17c509b8-9c7e-4149-ba6d-2c9652052c0f}, type=TaskType{typeName=memory-reference-task}, additionalInformation=Optional.empty, status=IN_PROGRESS, submittedDate=2019-09-24T13:25:30.429Z[GMT], submittedNode=Hostname(foo), startedDate=Optional[2019-09-24T13:25:30.512Z[GMT]], ranNode=Optional[Hostname(foo)], completedDate=Optional.empty, canceledDate=Optional.empty, cancelRequestedNode=Optional.empty, failedDate=Optional[2019-09-24T13:25:30.512Z[GMT]]}]>
[61962e3546564b6a83aa7ecf6d52676f4418c495] [INFO] 
[61962e3546564b6a83aa7ecf6d52676f4418c495] [ERROR] Tests run: 73, Failures: 1, Errors: 0, Skipped: 2
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**